core/src/io/anuke/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -281,9 +281,13 @@ public void drawLayer(Tile tile){
 
         for(int i = 0; i < entity.power.links.size; i++){
             Tile link = world.tile(entity.power.links.get(i));
-            if(linkValid(tile, link) && (link.pos() < tile.pos() || !(link.block() instanceof PowerNode) || !Core.camera.bounds(Tmp.r1).contains(link.drawx(), link.drawy()))){
```

Comment:
How does this handle cases where an extremely long-range node (e.g. surge tower) far off-screen is linking to a power generator that *is* on-screen, or vice versa? The whole reason power nodes are drawn twice is to fix potential issues of this sort.

Reply:
Anything on a screen that has a power link will schedule blocks that it links to for rendering in BlockRenderer. So for every connection (a-b) on screen tile a and b will go to drawLayer(Tile tile) of PowerNode. It least that was I figured out and tested.

Btw. power connections seemed to be drawn more than twice duo rescheduling same tile in BlockRenderer.